scripts/patch-spl-crates-for-anchor.sh

```diff
@@ -0,0 +1,59 @@
+    spl_tlv_account_resolution_verison=$(readCargoVariable version "$spl_dir/libraries/tlv-account-resolution/Cargo.toml")
+    spl_transfer_hook_interface_version=$(readCargoVariable version "$spl_dir/token/transfer-hook/interface/Cargo.toml")
+}
+
+patch_spl_crates() {
+    declare project_root="$1"
+    declare Cargo_toml="$2"
+    declare spl_dir="$3"
+    update_spl_dependencies "$project_root"
+    patch_crates_io "$Cargo_toml" "$spl_dir"
+}
+
+update_spl_dependencies() {
+    declare project_root="$1"
+    declare tomls=()
+    while IFS='' read -r line; do tomls+=("$line"); done < <(find "$project_root" -name Cargo.toml)
+
+    sed -i -e "s#\(spl-associated-token-account = \"\)[^\"]*\(\"\)#\1$spl_assocated_token_account_version\2#g" "${tomls[@]}" || return $?
+    sed -i -e "s#\(spl-associated-token-account = { version = \"\)[^\"]*\(\"\)#\1$spl_assocated_token_account_version\2#g" "${tomls[@]}" || return $?
+    sed -i -e "s#\(spl-memo = \"\)[^\"]*\(\"\)#\1$spl_memo_version\2#g" "${tomls[@]}" || return $?
+    sed -i -e "s#\(spl-memo = { version = \"\)[^\"]*\(\"\)#\1$spl_memo_version\2#g" "${tomls[@]}" || return $?
+    sed -i -e "s#\(spl-token = \"\)[^\"]*\(\"\)#\1$spl_token_version\2#g" "${tomls[@]}" || return $?
+    sed -i -e "s#\(spl-token = { version = \"\)[^\"]*\(\"\)#\1$spl_token_version\2#g" "${tomls[@]}" || return $?
+    sed -i -e "s#\(spl-token-2022 = \"\).*\(\"\)#\1$spl_token_2022_version\2#g" "${tomls[@]}" || return $?
+    sed -i -e "s#\(spl-token-2022 = { version = \"\)[^\"]*\(\"\)#\1$spl_token_2022_version\2#g" "${tomls[@]}" || return $?
+    sed -i -e "s#\(spl-tlv-account-resolution = \"\)[^\"]*\(\"\)#\1=$spl_tlv_account_resolution_verison\2#g" "${tomls[@]}" || return $?
+    sed -i -e "s#\(spl-tlv-account-resolution = { version = \"\)[^\"]*\(\"\)#\1=$spl_tlv_account_resolution_verison\2#g" "${tomls[@]}" || return $?
+    sed -i -e "s#\(spl-transfer-hook-interface = \"\)[^\"]*\(\"\)#\1=$spl_transfer_hook_interface_version\2#g" "${tomls[@]}" || return $?
+    sed -i -e "s#\(spl-transfer-hook-interface = { version = \"\)[^\"]*\(\"\)#\1=$spl_transfer_hook_interface_version\2#g" "${tomls[@]}" || return $?
+
+    # patch ahash. This is super brittle; putting here for convenience, since we are already iterating through the tomls
+    ahash_minor_version="0.8"
+    sed -i -e "s#\(ahash = \"\)[^\"]*\(\"\)#\1$ahash_minor_version\2#g" "${tomls[@]}" || return $?
+}
+
+patch_crates_io() {
+    declare Cargo_toml="$1"
+    declare spl_dir="$2"
+    cat >> "$Cargo_toml" <<EOF
+    spl-memo = { path = "$spl_dir/memo/program" }
+    spl-token = { path = "$spl_dir/token/program" }
+    spl-token-2022 = { path = "$spl_dir/token/program-2022" }
+    spl-tlv-account-resolution = { path = "$spl_dir/libraries/tlv-account-resolution" }
+    spl-transfer-hook-interface = { path = "$spl_dir/token/transfer-hook/interface" }
+EOF
+}
```

Comment:
This seems to be missing `spl-associated-token-account` -- is that intentional?

Reply:
Oops, no, not intentional. I added ATA at the 11th hour for completeness, although it works fine without patching it, and didn't do a thorough job. Fixed.

Reply:
Oh yeah, that's why I didn't patch ATA: it breaks everything 😞 
How would you feel about merging this with the ATA patch reverted, and then I'll figure out why it breaks later?

Reply:
Yeah that's fine, sorry for causing the issue!

Reply:
It may be related to the fact that SPL downstream jobs are now broken again over here: https://github.com/anza-xyz/agave/pull/121

I'm seeing 2 versions of ATA in my Cargo.lock when I run SPL patch locally:
```
[[package]]
name = "spl-associated-token-account"
version = "2.3.1"
dependencies = [..]

[[package]]
name = "spl-associated-token-account"
version = "2.3.1"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "4414117bead33f2a5cf059cefac0685592bdd36f31f3caac49b89bff7f6bbf32"
dependencies = [..]
```

Maybe-hopefully when I fix that, it provides clarity back here as well.

Reply:
Hm, that's very strange